Documentation/Proposals/NNNN-public-issues-api.md

```diff
@@ -0,0 +1,178 @@
+# Public Issues API
+
+* Proposal: [SWT-NNNN](NNNN-public-issues-api.md)
+* Authors: [Rachel Brindle](https://github.com/younata)
+* Status: **Awaiting review**
+* Bug: [apple/swift-testing#474](https://github.com/apple/swift-testing/issues/474)
+* Implementation: [apple/swift-testing#481](https://github.com/apple/swift-testing/pull/481)
+* Review: n/a
+
+## Introduction
+
+Swift Testing should provide a more robust Issue-reporting API, to facilitate
+better integration with third-party assertion tools.
+
+## Motivation
+
+There are several third-party assertion libraries that developers are used to
+using, and Swift Testing should make it as easy as possible for the developers
+and maintainers of these third-party assertion libraries to integrate with
+Swift Testing.
+
+Swift Testing currently offers the `Issue.record` family of static methods,
+which provides an acceptable interface for one-off/custom assertions without
+using the `#expect` or `#require` macros. Unfortunately, `Issue.record` is not
+robust enough for dedicated assertion libraries like Nimble. For example,
+there is no current way to provide an `Issue.Kind` to `Issue.record`.
+
+
+## Proposed solution
+
+I propose making the initializer for `Issue` public, and provide a public
+`Issue.record()` instance method to report that Issue. This provides a clean,
+easy to maintain way to create and record custom Issues that assertion tools
+are already used to because it is analogous to the XCTIssue API in XCTest.
+
+To help support this, I also propose making a public initializer for
+`Expectation`. This public initializer will not take in an `Expression`.
+Instead, the public initializer will create an empty `Expression`, in order to
+maintain compatibility with the current JSON ABI.
+
+## Detailed design
+
+Create a public initializer for `Issue`, which is not gated by
+`@_spi(ForToolsIntegrationOnly)`:
+
+```swift
+public struct Issue: Sendable {
+  // ...
+
+  /// Initialize an issue instance with the specified details.
+  ///
+  /// - Parameters:
+  ///   - kind: The kind of issue this value represents.
+  ///   - comments: An array of comments describing the issue. This array may
+  ///     be empty.
+  ///   - sourceContext: A ``SourceContext`` indicating where and how this
+  ///     issue occurred. This defaults to a default source context returned by
+  ///     calling ``SourceContext/init(backtrace:sourceLocation:)`` with zero
+  ///     arguments.
+  public init(
+    kind: Kind,
+    comments: [Comment] = [],
+    sourceContext: SourceContext = .init()
+  ) {}
```

Comment:
Proposals don't need to include the bodies of methods/initializers/etc. You can leave the squiggly brackets out here.